.github/workflows/unit-test.yml

```diff
@@ -135,43 +110,20 @@ jobs:
           python-version: '${{ inputs.python_version }}'
 
       - name: Upgrade PIP
-        if: steps.continue.outputs.confirm == 'true'
-        run: pip install --upgrade pip && pip install wheel --upgrade
-
-      - name: Install PennyLane dependencies
         if: steps.continue.outputs.confirm == 'true'
         run: |
-          pip install -r requirements-ci.txt --upgrade
-          pip install -r requirements-dev.txt --upgrade
-
-      - name: Install PyTorch
-        if: inputs.install_pytorch == true && steps.continue.outputs.confirm == 'true'
-        env:
-          TORCH_VERSION: ${{ inputs.pytorch_version != '' && format('=={0}', inputs.pytorch_version) || '' }}
-        run: pip install "torch${{ env.TORCH_VERSION }}" -f https://download.pytorch.org/whl/torch_stable.html
-
-      - name: Install TensorFlow
-        if: inputs.install_tensorflow == true && steps.continue.outputs.confirm == 'true'
-        env:
-          TF_VERSION: ${{ inputs.tensorflow_version != '' && format('~={0}', inputs.tensorflow_version) || '' }}
-        run: pip install "tensorflow${{ env.TF_VERSION }}" "keras${{ env.TF_VERSION }}"
+          pip install --upgrade pip
+          pip install --upgrade poetry wheel
+          poetry config virtualenvs.create false
```

Comment:
I would highly recommend usage, of an action that can setup poetry. It's being used in QML as well [See Here](https://github.com/PennyLaneAI/qml/blob/6b3d4ca24970b12f3be7814e62a7749d0f41aaec/.github/workflows/build-branch.yml#L253-L258)

Link to the action itself: https://github.com/XanaduAI/cloud-actions/tree/main/install-python-poetry

The benefit of using the action is that it installs poetry into a separate env to ensure poetry never attempts to manage it's own dependencies.

Reply:
already a cloud-action 😍 i'll give it a shot